### PR TITLE
ISSUE-1441: add support to git clone recursive submodules

### PR DESCRIPTION
**Pull request by jgrasser**

### CODE REVIEW

**Comment by insspb** [comment]
@jgrasser thank you for contribution....

i am just returning to work on cc. 

Several questions, regarding this pull request.

1. hg has no recursive option?
2. Why not as cli option to cc? It is possible that final user do not need to make submodules clone.

Maybe think about passing user commands to hg/git in some way? So user decide what he want...

**Comment by faph** [comment]
For cookiecutters, I wonder whether `--recurse-submodules` should be the default? Would there be any scenario where one would create a cookiecutter repo and *not* intend the sub-modules to be part of the baked project? Happy for it to be configurable, but would prefer if it became an opt-out thing.

**Comment by gliptak** [comment]
https://www.mercurial-scm.org/wiki/Subrepository

**Comment by JonZeolla** [comment]
@jgrasser did you see the questions above?

**Comment by jgrasser** [comment]
Hey, just catching up on this now. Thank you all for the feedback. :)

I don't have any mercurial experience; I'm one of those people that left university right when git was introduced and I never attempted to broaden my horizons by it.

Is it correct that `hg clone` will do the recurse clone by default? Also, to checkout/pull the submodules you just need to append the `-s` flag to the mercurial commands?

File: cookiecutter/main.py [hunk 1]
```diff
@@ -39,6 +40,8 @@ def cookiecutter(
     :param template: A directory containing a project template directory,
         or a URL to a git repository.
     :param checkout: The branch, tag or commit ID to checkout after clone.
+    :param checkout: The branch, tag or commit ID to checkout after clone.
```
Comment (JonZeolla) [inline cookiecutter/main.py:43]: Duplicate

**Comment by yuvaldolev** [comment]
Hi @jgrasser @insspb @faph @gliptak @JonZeolla 

We started working on a new community fork of the Cookiecutter project, named Cookieninja.
The fork is found at: https://github.com/cookieninja-generator/cookieninja

We have re-opened your PR in the Cookieninja project at: https://github.com/cookieninja-generator/cookieninja/pull/51

We're trying to bring the life back to this amazing project.
We are committed to reviewing and merging each PR in up to 2 weeks from the day it's opened (if there are no issues with it).
We'd really like you to join the fork, star it and contribute to it!

Thank you :)

**Pull request closed without merging**